### PR TITLE
[DDC-3378] Support merging entities with composite identities defined through to-one associations

### PR DESCRIPTION
**Pull request by adrienbrault**
I hope that I'm doing something wrong and that this is not a bug ...

### CODE REVIEW

**Comment by Ocramius** [comment]
@hjr3 does this apply also to the codebase after @adrienbrault's patch?

**Comment by hjr3** [comment]
@Ocramius the patch from @adrienbrault does address my issue.

**Comment by Ocramius** [comment]
@adrienbrault can you check @stof's comment please?

File: lib/Doctrine/ORM/Id/AssignedGenerator.php [hunk 1]
```diff
@@ -59,12 +59,8 @@ public function generate(EntityManager $em, $entity)
             }
 
             if (isset($class->associationMappings[$idField])) {
-                if ( ! $em->getUnitOfWork()->isInIdentityMap($value)) {
-                    throw ORMException::entityMissingForeignAssignedId($entity, $value);
-                }
```
Comment (Ocramius) [inline lib/Doctrine/ORM/Id/AssignedGenerator.php]: I think this is actually correct though - identities deriving from associations are still valid, even for un-managed entities.

Comment (stof) [inline lib/Doctrine/ORM/Id/AssignedGenerator.php]: hmm, indeed. The relation itslef will still be required to use a managed entity in other places

**Comment by Ocramius** [comment]
@adrienbrault merged, thanks!

`master`: doctrine/doctrine2@b889e18a9a15c36eac7349b06cb4b0f955a9da57

**Pull request closed without merging**